src/object/cli_obj.c

```diff
@@ -147,7 +147,8 @@ obj_alloc(void)
 void
 obj_decref(struct dc_object *obj)
 {
-	daos_hhash_link_putref(&obj->cob_hlink);
+	if (obj != NULL)
```

Comment:
not your problem for this PR, but this better return DER_INVAL if not valid.

Reply:
maybe assertion() ?

Reply:
In TX logic, the obj_decref() will be called during failure handling, under some cases, the "obj" may be NULL. In fact, the caller can check whether "obj" is NULL or not before calling obj_decref(), but that will make more code lines. So I move the check inside obj_decref(). Similar logic as "D_FREE()" case which caller does not need to care about whether the input "ptr" is NULL or not, instead, it is the D_FREE() itself to check whether input "ptr" is valid or not.

Reply:
>In fact, the caller can check whether "obj" is NULL or not before calling obj_decref(), but that will make more code lines.

that is why adding assertion in obj_decref()? which should not caught any failure in CI tests?